clients/mocks.go

```diff
@@ -8,33 +8,50 @@ type (
 	GatekeeperMock struct {
 		expectedPermissions Permissions
 		expectedError       error
+		UserIDs             []string
 	}
 	SeagullMock struct{}
 )
 
 //A mock of the Gatekeeper interface
 func NewGatekeeperMock(expectedPermissions Permissions, expectedError error) *GatekeeperMock {
-	return &GatekeeperMock{expectedPermissions, expectedError}
+	return &GatekeeperMock{expectedPermissions, expectedError, []string{}}
+}
+func (mock *GatekeeperMock) SetExpected(expectedPermissions Permissions, expectedError error) {
```

Comment:
although this is straightforward, shall we have a unit test for that too?

Reply:
The test should be implemented if we do more stuff than just assigning 2 private values... 
For now, I think we can continue without testing it

Reply:
ok for me